mylyn.commons/org.eclipse.mylyn.commons.identity-feature/p2.inf

```diff
@@ -0,0 +1,6 @@
+update.id=org.eclipse.mylyn.commons.identity.feature.group
+update.range=[0,4)
+update.matchExp = providedCapabilities.exists(pc | \
+   pc.namespace == 'org.eclipse.equinox.p2.iu' && \
+     (pc.name == 'org.eclipse.mylyn.commons.identity.feature.group' || \
+       (pc.name == 'org.eclipse.mylyn.commons.identity.feature.group' && pc.version ~= range('[0.0.0,$version$)'))))
```

Comment:
Now that I'm how and look at this more closely on the big monitor, I wonder if it's correct?

Here the second condition with the range is the old name:

https://github.com/eclipse-mylyn/org.eclipse.mylyn/blob/2cf887a050aaffec93ee1999ffc84f97e14c17cc/mylyn.egit/org.eclipse.mylyn.egit.feature/p2.inf#L6

So should this not be org.eclipse.mylyn.commons.identity**.feature.**feature.group?  Also is then the update.id wrong and should be the old name too?

If this is a mistake, is it repeated elsewhere?

Reply:
In the end, we should test that a 2023-06 installation with all of Mylyn's features installed can be updated to you nightly build with the new version IDs and these p2.inf instructions...

Reply:
What was done for egit is
```
update.id=old.feature.group
update.range=[0,4)
update.matchExp = providedCapabilities.exists(pc | \
   pc.namespace == 'org.eclipse.equinox.p2.iu' && \
     (pc.name == 'new.feature.group' || \
       (pc.name == 'old.feature.group' && pc.version ~= range('[0.0.0,$version$)'))))
```
now I changed it accordingly (and I'm going to revisit all the merged `p2.inf` files when we can converge on this one).
But, what I can see [here](https://wiki.eclipse.org/Equinox/p2/Customizing_Metadata#Update_descriptor_advice:) is a bit different:
```
update.matchExp = providedCapabilities.exists(pc | \
   pc.namespace == 'org.eclipse.equinox.p2.iu' && \
     (pc.name == 'old.feature.group' || \
       (pc.name == 'new.feature.group' && pc.version ~= range('[0.0.0,$version$)'))))
```
Should I use [egit](https://github.com/eclipse-mylyn/org.eclipse.mylyn/blob/2cf887a050aaffec93ee1999ffc84f97e14c17cc/mylyn.egit/org.eclipse.mylyn.egit.feature/p2.inf#L6) as a working sample Ed @merks ?

Reply:
I think the egit one was tested, but my memory is not as good as it once was...

Note that one can easily build your update sites locally and test such things locally.  I was doing that just now to look at deleting the uncategorized category...

Reply:
> Note that one can easily build your update sites locally and test such things locally.

I'm sure one can. Especially if he's not the only one working on the project :)

Anyway, there is no alternative, I'll try it on my side.

Reply:
My findings so far:
1) snippet from [here](https://wiki.eclipse.org/Equinox/p2/Customizing_Metadata#Update_descriptor_advice:) looks right and logical, with one change: we need to use exact version value instead of `$version$` macro, otherwise condition will not work for us in the next version.
```
update.matchExp = providedCapabilities.exists(pc | \
   pc.namespace == 'org.eclipse.equinox.p2.iu' && \
     (pc.name == 'old.feature.group' || \
       (pc.name == 'new.feature.group' && pc.version ~= range('[0.0.0,4.0.0)'))))
```
2) `update.id=old.feature.group` and `update.range=[0,4)` are not needed, since they will be ignored anyway

3) the most of already renamed features was recognized for update after changes described above in this comment. For the rest of them I suspect the `*SDK` features who trouble us.
![image](https://github.com/eclipse-mylyn/org.eclipse.mylyn/assets/15242312/ad1d3706-3360-415d-bc50-d6dda0fe3ed9)

Reply:
Should I add other revised `p2.inf` resources to this PR or provide another one @merks ?

Reply:
Do what's easiest and fastest for you personally!

Reply:
> Do what's easiest and fastest for you personally!

I cannot proceed without approvals and I have no idea what to do with it. 
Personally I prefer small PRs, because it is easier to review them. But since time to collect review approvals is unpredictable I can combine all the changes in the one big PR. 
Since you are one of the most active reviewers here in Mylyn I'm asking what is more convenient for you Ed @merks

Reply:
I hate to not answer your question.  I see emails quickly and generally react to them quickly.  So small PRs are fine.  But one big one is fine too.  I should note that Oomph's build has been broken (broken target platform) every night by these slowly arriving incremental changes so I would hope to have it done sooner rather than later or with a big bang only for that reason.  😱

Reply:
Why don't we switch off the requirement for review for now?

Reply:
Yes, I expect everyone is well behaved and will know what changes are controversial such that a review is requested as needed.

Reply:
Exactly.

Reply:
> Why don't we switch off the requirement for review for now?

We can do it via Helpdesk, and I did it for short period before 3.26.1. 
But IT people have concerns since we still have a lot of inactive committers.

Reply:
That makes sense. I will retire them.

Reply:
> Do what's easiest and fastest for you personally!

I'll provide the rest of work required for Eclipse Mylyn 4.0.0 M1 here, since I don't see another option to catch up the SimRel train

Reply:
The runway is short. If you need, I can delay until late morning.  But you need to let me know.

Reply:
Now I need to assembly milestone build and fix our SimRel contribution manifest